### PR TITLE
[Warlock] Refactors

### PR DESCRIPTION
**Pull request by sMteX**
Mostly minor refactors:
- extraction of constants before class definition (FatalEchoes, SoulShardDetails in all specs etc.)
- trying to eliminate computations in `on_finished()` where unnecessary (Empowered Life Tap modules, Immolate uptime etc.)
- safety checks for division by zero (UABuffTracker)
- unified the text for item damage (SacrolashsDarkStrike, StretensSleeplessShackles etc., it's possible I missed something)
- minor text cleanup (PowerCordOfLethtendris, FeretoryOfSouls)
- refactor of SoulHarvest modules - private variables/methods renames and using `on_byPlayerPet_damage`
- empty lines here and there for better readability of the code (or to logically separate some parts)

Last but not least, I gave a second thought to Fire and Brimstone and realized it's way too unnecessarily complex and reduced it down. As mentioned in the commit, it does give different results than the previous implementation, but it seems that it's more likely the previous one was wrong and not this one.

Should not cause merge conflicts with #432 but if yes, they shouldn't be hard to resolve.

### CODE REVIEW

File: src/Parser/AfflictionWarlock/Modules/Features/UABuffTracker.js [hunk 1]
```diff
@@ -72,7 +72,7 @@ class UABuffTracker extends Module {
   }
 
   suggestions(when) {
-    const unbuffedTicksPercentage = this.unbuffedTicks / this.totalTicks;
+    const unbuffedTicksPercentage = this.unbuffedTicks / (this.totalTicks > 0 ? this.totalTicks : 1);
```
Comment (kfinch) [inline src/Parser/AfflictionWarlock/Modules/Features/UABuffTracker.js]: Cleaner to do `const unbuffedTicksPercentage = (this.unbuffedTicks / this.totalTicks) || 1;`

File: src/Parser/AfflictionWarlock/Modules/Features/UABuffTracker.js [hunk 2]
```diff
@@ -84,7 +84,7 @@ class UABuffTracker extends Module {
   }
 
   statistic() {
-    const buffedTicksPercentage = 1 - (this.unbuffedTicks / this.totalTicks);
+    const buffedTicksPercentage = 1 - (this.unbuffedTicks / (this.totalTicks > 0 ? this.totalTicks : 1));
```
Comment (kfinch) [inline src/Parser/AfflictionWarlock/Modules/Features/UABuffTracker.js]: Cleaner to do `const unbuffedTicksPercentage = (this.unbuffedTicks / this.totalTicks) || 1;`

File: src/Parser/AfflictionWarlock/Modules/Items/Tier20_4set.js [hunk 3]
```diff
@@ -34,8 +34,8 @@ class Tier20_4set extends Module {
       icon: <SpellIcon id={SPELLS.WARLOCK_AFFLI_T20_4P_BONUS.id} />,
       title: <SpellLink id={SPELLS.WARLOCK_AFFLI_T20_4P_BONUS.id} />,
       result: (
-        <dfn data-tip={'This result may be inaccurate as effect of haste buffs is difficult to calculate. This result is based on the premise that 15% haste allows you to do <code>(baseHaste * (100% + 15%) + 15%) - baseHaste</code> more things in the same amount of time, therefore the set bonus is estimated to be technically 15% damage increase for 8 seconds.'}>
-          {`${formatNumber(this.bonusDmg)} damage - ${this.owner.formatItemDamageDone(this.bonusDmg)}`}
+        <dfn data-tip={`Estimated bonus damage contributed: ${formatNumber(this.bonusDmg)}.<br /><br />This result may be inaccurate as effect of haste buffs is difficult to calculate. This result is based on the premise that 15% haste allows you to do <code>(baseHaste * (100% + 15%) + 15%) - baseHaste</code> more things in the same amount of time, therefore the set bonus is estimated to be technically 15% damage increase for 8 seconds.`}>
+          {this.owner.formatItemDamageDone(this.bonusDmg)}
```
Comment (kfinch) [inline src/Parser/AfflictionWarlock/Modules/Items/Tier20_4set.js]: In cases where you're providing an acknowledged estimate of throughput, I think it helps to also put in the tooltip the closest non-estimated stat you have. In this case, might be nice to show the buffs percent uptime.

**Comment by sMteX** [comment]
Pushed the minor changes and changed the Affli T20 4p module. I didn't like extrapolating so much from the beginning when I wrote that module so I'll just show the uptime instead, that's at least correct.

**Comment by kfinch** [comment]
👍

**Pull request merged**